### ISSUE TITLE
No way to exclude directories from inotify watching

### ISSUE THREAD

**Issue by akdor1154** [spyder-ide/spyder#10664]
Feature request: a way to exclude directories from inotify watches, with sensible defaults.

Currently Spyder is using a full half of the default 8192 inotify watches on my Ubuntu system. This is in a single project with less than 30 python files. I assume what is happening is the .venv directory in my project is being watched, along with possibly __pycache__ and .mypy_cache directories. It would be great if a setting could be exposed to let me exclude these directories from being watched. As a bonus, these directories (+ maybe `.virtualenv`, `virtualenv`, etc?) could probably be excluded by default.

I am aware that the limit of watches can be increased in my system's config, but in this case there is no reason for Spyder to be using so many watches in the first place.

**Comment by ccordoba12** [comment]
> It would be great if a setting could be exposed to let me exclude these directories from being watched. As a bonus, these directories (+ maybe .virtualenv, virtualenv, etc?) could probably be excluded by default.

I agree. There's no need to track files in those directories, plus the ones users don't want to, for whatever reason.

We'll take a look at this after we release Spyder 4 because (unfortunately) it came up very late to our attention.

**Comment by akdor1154** [comment]
I've had a quick look at this and the lib you are using, `watchdog`, does not support this as-is (requested feature since 2014...). However, it might be possible to subclass one of watchdog's classes to implement this. Would you guys consider such a solution if it wasn't too ugly? Or would you insist that this is implemented in `watchdog` itself first?

**Comment by ccordoba12** [comment]
> Would you guys consider such a solution if it wasn't too ugly? 

No. If you're willing to submit a PR for this, we'd really appreciate it!

**Comment by goanpeca** [comment]
>However, it might be possible to subclass one of watchdog's classes to implement this. Would you guys consider such a solution if it wasn't too ugly?

It is what usually needs to be done with Watchdog @akdor1154 so that would be fine :+1:
